src/eth/eip_155_transaction.rs

```diff
@@ -36,6 +42,110 @@ pub struct EthLegacyEip155TxArgs {
 }
 
 impl EthLegacyEip155TxArgs {
+    pub fn signature(&self, eth_chain_id: EthChainId) -> anyhow::Result<Signature> {
+        // Validate EIP155 Chain ID
+        if let Err(err) = validate_eip155_chain_id(eth_chain_id, &self.v) {
+            bail!("failed to validate EIP155 chain id: {}", err);
+        }
+
```

Comment:
```suggestion
        validate_eip155_chain_id(eth_chain_id, &self.v)?;
```
I don't think another level brings any value here.